shell/platform/windows/flutter_windows_unittests.cc

```diff
@@ -149,5 +149,50 @@ TEST_F(WindowsTest, VerifyNativeFunctionWithReturn) {
   EXPECT_TRUE(bool_value_passed);
 }
 
+// Verify the next frame callback is executed.
+TEST_F(WindowsTest, NextFrameCallback) {
+  struct Captures {
+    fml::AutoResetWaitableEvent latch;
+    std::thread::id thread_id;
+  };
+  Captures captures;
+
+  CreateNewThread("test_platform_thread")->PostTask([&]() {
+    captures.thread_id = std::this_thread::get_id();
+
+    auto& context = GetContext();
+    WindowsConfigBuilder builder(context);
+    builder.SetDartEntrypoint("drawHelloWorld");
+
+    ViewControllerPtr controller{builder.Run()};
+    ASSERT_NE(controller, nullptr);
+
+    auto engine = FlutterDesktopViewControllerGetEngine(controller.get());
+
+    FlutterDesktopEngineSetNextFrameCallback(
+        engine,
+        [](void* user_data) {
+          auto captures = static_cast<Captures*>(user_data);
+
+          // Callback should execute on platform thread.
+          ASSERT_EQ(std::this_thread::get_id(), captures->thread_id);
+
+          // Signal the test passed and end the Windows message loop.
+          captures->latch.Signal();
+          ::PostQuitMessage(0);
+        },
+        &captures);
+
+    // Pump messages for the Windows platform runner.
+    ::MSG msg;
+    while (::GetMessage(&msg, nullptr, 0, 0)) {
+      ::TranslateMessage(&msg);
+      ::DispatchMessage(&msg);
+    }
```

Comment:
This is necessary as otherwise platform tasks won't run. The Windows platform thread's task runner is [actually a window](https://github.com/flutter/engine/blob/b4cbebe3887e7cc1f0a6f8646856bc1f1cdde53e/shell/platform/windows/task_runner_window.cc#L15-L17). It is woken up by [posting a message to this window](https://github.com/flutter/engine/blob/b4cbebe3887e7cc1f0a6f8646856bc1f1cdde53e/shell/platform/windows/task_runner_window.cc#L54-L58). If messages aren't dispatched, the Windows platform task runner is never woken up, and platform tasks are never executed.